File: mvp/ts/src/parse.ts
import { DeviceReading } from "./models";
import { renameKeys, unitConvert, enumMap } from "./transforms";

export function parseDevicePayload(raw: any){
  let x = renameKeys(raw);
  x = unitConvert(x);
  x = enumMap(x);
  return DeviceReading.parse(x);
}

File: mvp/python/app/parse.py
from .models import DeviceReading
from .transforms import rename_keys, unit_convert, enum_map

def parse_device_payload(raw: dict) -> DeviceReading:
    x = rename_keys(raw)
    x = unit_convert(x)
    x = enum_map(x)
    return DeviceReading.model_validate(x)
